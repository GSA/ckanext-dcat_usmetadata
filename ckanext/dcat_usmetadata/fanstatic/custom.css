/* Hide original "Add dataset" button */
.page_primary_action > a.btn-primary:nth-of-type(2) {
    display: none !important;
}
