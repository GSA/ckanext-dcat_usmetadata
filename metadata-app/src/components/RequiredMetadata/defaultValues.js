export default {
  access_level_comment: 'true',
  spatial: 'false',
  spatial_location: 'false',
  temporal: 'false',
  modifiedOther: '',
  publisher: '',
};
