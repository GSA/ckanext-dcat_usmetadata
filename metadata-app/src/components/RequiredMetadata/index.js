import React, { useState } from 'react';
import PropTypes from 'prop-types';
import WrappedField from '../WrappedField';
import TagsAutocomplete from '../TagsAutocomplete';
import { ReactComponent as Info } from '../../img/info.svg';
import HelpText from '../HelpText';
import Radio from '../Radio';

const RequiredMetadata = (props) => {
  const { values, errors, apiUrl, apiKey } = props;

  // RADIO / SELECT Values
  const [urlDisabled, setUrlDisabled] = useState(true);
  const [toolTipShown, setToolTipShown] = useState(false);
  // eslint-disable-next-line
  const toggleToolTip = () => setToolTipShown(toolTipShown ? false : true);

  const urlify = (text) => {
    return text ? text.replace(/\s+/g, '-').toLowerCase() : '';
  };

  const helpTextify = (text) => {
    return <HelpText>{text}</HelpText>;
  };
  const baseUrl = `${window.location.origin}/dataset/`;

  const helpTexts = {
    title: (
      <HelpText>
        Use <a href="https://plainlanguage.gov/">everyday language</a> to make the dataset easy to
        find and understand
      </HelpText>
    ),
    description: (
      <HelpText>
        Write a description (like an abstract) with enough detail to help a user quickly decide if
        the asset is of interest. You can use{' '}
        <a href="https://www.markdownguide.org/basic-syntax/">Markdown Formatting</a> here.
      </HelpText>
    ),
    select: (
      <HelpText>
        If you do not see the Publisher for your dataset listed, please contact{' '}
        <a href="mailto:inventory-help@gsa.gov">inventory-help@gsa.gov</a> for further assistance.
        Start typing to add tags.
      </HelpText>
    ),
  };

  return (
    <div className="usa-form-custom">
      <section id="section-basic-mega-menu" className="site-component-section">
        <p className="site-text-intro">
          The following fields are required metadata for each dataset in an agency’s inventory (
          <a href="https://www.congress.gov/115/plaws/publ435/PLAW-115publ435.pdf">
            per Section 202 of the OPEN Government Data Act)
          </a>{' '}
          For more information about the form fields, consult the
          <a href="https://resources.data.gov/resources/dcat-us/">DCAT-US Schema.</a>
        </p>
      </section>
      <div className="row">
        <div className="grid-col-12">
          <WrappedField
            label="Title"
            name="title"
            type="string"
            placeholder=""
            helptext={helpTexts.title}
            value={values.title}
            errors={errors}
            required
          />

          <div>
            <WrappedField
              name="url"
              type="string"
              style={{ display: urlDisabled ? 'none' : 'inline' }}
              value={values.url || `${baseUrl}${urlify(values.title)}`}
              errors={errors}
            />
            <span className="dataset_url" style={{ display: urlDisabled ? 'inline' : 'none' }}>
              {`${baseUrl}${urlify(values.title)}`}
            </span>

            <button
              type="button"
              className="usa-button dataset_url_edit"
              style={{ display: urlDisabled ? 'inline' : 'none' }}
              onClick={() => {
                setUrlDisabled(false);
              }}
            >
              Edit
            </button>
          </div>
        </div>
      </div>
      <div className="row">
        <WrappedField
          label="Description"
          name="description"
          type="string"
          component="textarea"
          rows="6"
          helptext={helpTexts.description}
          value={values.description}
          errors={errors}
          required
        />
      </div>
      <div className="row">
        <span className="usa-label">Tags*</span>
        <TagsAutocomplete
          id="tags-autocomplete-input"
          tags={values.tags}
          apiUrl={apiUrl}
          apiKey={apiKey}
          name="tags"
          titleField="name"
          required
          placeholderText="Start typing to search"
          errors={errors}
          helptext={helpTextify(
            'Use both technical and non-technical terms to help users find your dataset.'
          )}
        />
      </div>
      <div className="row">
        <WrappedField
          label="Publisher"
          name="publisher"
          type="select"
          choices={['Publisher 1 ', 'Publisher 2', 'Publisher 3', 'Publisher 4']}
          required
          className="error-msg"
          helptext={helpTexts.select}
          infoText="The publishing entity (e.g. your agency) and optionally their parent organization(s)."
          errors={errors}
        />
      </div>
      <div className="row">
        <WrappedField
          label="Sub Agency"
          name="subagency"
          type="select"
          choices={['Sub Agency 1 ', 'Sub Agency 2', 'Sub Agency 3', 'Sub-Agency 4']}
          className="error-msg"
          errors={errors}
        />
      </div>
      <div className="row">
        <WrappedField
          label="Contact Name"
          name="contactPoint"
          type="string"
          required
          infoText="This should be the person who can best answer or triage questions about this dataset, either on the metadata or the substance of the data resources."
          errors={errors}
        />
      </div>
      <div className="row">
        <WrappedField
          label="Contact Email"
          name="contactEmail"
          type="string"
          required
          errors={errors}
        />
      </div>
      <div className="row">
        <WrappedField label="Unique ID" name="identifier" type="string" required errors={errors} />
      </div>
      <div className="row">
        <WrappedField
          label="Public Access level"
          name="accessLevel"
          type="select"
          choices={['public', 'restricted public', 'non-public']}
          className="error-msg"
          errors={errors}
          required
        />
      </div>
      <div className="row">
        <WrappedField
          label="Meets Agency Data Quality"
          name="dataQuality"
          type="select"
          choices={['Yes', 'No']}
          className="error-msg"
          errors={errors}
          required
        />
      </div>
      <div className="row">
        <WrappedField
          label="License"
          name="license"
          type="select"
          choices={['MIT', 'Open Source License', 'Others']}
          className="error-msg"
          errors={errors}
          required
        />
        <WrappedField
          name="license_others"
          type="string"
          helptext={helpTextify(
            `If you selected “Other”, please specify the name of your License*'`
          )}
          disabled={values.license !== 'Others'}
          errors={errors}
          required
        />
      </div>
      <div className="row">
        <span className="usa-label">Rights</span> <br />
        {errors && errors.rights && <span className="error-msg">{errors.rights}</span>}
        <Radio
          label="My dataset is public"
          name="rights"
          errors={errors}
          value="true"
          id="rights_option_1"
        />
        <Radio label="My dataset is not public" name="rights" value="false" id="rights_option_2" />
        <WrappedField
          name="rights_desc"
          type="string"
          value={values.rights_desc}
          errors={errors}
          helptext={helpTextify(
            'If your dataset is not public, please add an explanation of rights and feel free to include any instructions on restrictions, or how to access a restricted file (max 255 characters)*'
          )}
          disabled={values.rights === 'true'}
        />
      </div>

      <div className="row">
        <span className="usa-label">Relevant Location*</span> <br />
        {errors && errors.spatial && <span className="error-msg">{errors.spatial}</span>}
        <Radio
          label="My dataset does not have a spatial component"
          name="spatial"
          value="false"
          errors={errors}
          id="spatial_option_1"
        />
        <Radio
          label="My dataset does have a spatial component"
          name="spatial"
          value="true"
          id="spatial_option_2"
        />
        <WrappedField
          name="spatial_location_desc"
          type="string"
          value={values.spatial_location_desc}
          errors={errors}
          helptext={helpTextify(
            'If your dataset has a spatial component, please provide location such as place name or latitude/longitude pairs above*'
          )}
          disabled={values.spatial === 'false'}
        />
      </div>

      <div className="row">
        <span className="usa-label">
          Temporal*
          <div className={`tooltip ${toolTipShown ? 'show' : ''}`}>
            <Info
              tabIndex={0}
              height="20px"
              width="20px"
              style={{ marginLeft: '.5em' }}
              onClick={() => toggleToolTip()}
              onKeyUp={(e) => {
                if (e.keyCode === 13) {
                  toggleToolTip();
                }
              }}
            />
            <span className="tooltiptext">
              <span
                tabIndex={0}
                className="close"
                onClick={() => setToolTipShown(false)}
                role="button"
                onKeyDown={() => setToolTipShown(false)}
              >
                <span className="close-tag">&times;</span>
              </span>
              <h3>Temporal</h3>
              <p>
                For example, for a 2010 Census dataset, the temporal extent would cover a period of
                time beginning 2000-04-02 and ending 2010-04-01.
              </p>
            </span>
          </div>
        </span>{' '}
        <br />
        {errors && errors.temporal && <span className="error-msg">{errors.temporal}</span>}
        <Radio
          label="My dataset does not have a start and end date for the applicability of data"
          name="temporal"
          value="false"
          errors={errors}
          id="temporal_option_1"
        />
        <Radio
          label="My dataset has a start and end date for the applicability of data"
          name="temporal"
          value={values.temporal}
          errors={errors}
          id="temporal_option_2"
        />
        <WrappedField
          name="temporal_start_date"
          type="date"
          id="temporal_start_date"
          value={values.temporal_start_date}
          errors={errors}
          helptext={helpTextify(
            'If your dataset has a temporal component, please provide start date for applicability of data above*'
          )}
          disabled={values.temporal === 'false'}
        />
        <WrappedField
          name="temporal_end_date"
          type="date"
          id="temporal_end_date"
          value={values.temporal_end_date}
          errors={errors}
          helptext={helpTextify(
            'If your dataset has a temporal component, please provide start date for applicability of data above*'
          )}
          disabled={values.temporal === 'false'}
        />
      </div>

      <div className="row">
        <button type="button" className="usa-button usa-button--outline">
          Save as draft
        </button>
        <button className="usa-button" type="submit">
          Save and Continue
        </button>
      </div>
    </div>
  );
};

RequiredMetadata.propTypes = {
  apiUrl: PropTypes.string.isRequired,
  apiKey: PropTypes.string.isRequired,
  values: PropTypes.any, // eslint-disable-line
  errors: PropTypes.any, // eslint-disable-line
};

export default RequiredMetadata;
