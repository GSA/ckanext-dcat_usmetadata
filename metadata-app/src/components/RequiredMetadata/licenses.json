[
  {
    "label": "N/A - My dataset doesnâ€™t have a license",
    "value": "n/a"
  },
  {
    "label": "MIT",
    "value": "https://opensource.org/licenses/MIT-0"
  },
  {
    "label": "U.S. Government Works",
    "value": "https://www.usa.gov/publicdomain/label/1.0/"
  },
  {
    "label": "Creative Commons Zero Public Domain Dedication (CC0)",
    "value": "https://creativecommons.org/publicdomain/zero/1.0/"
  },
  {
    "label": "Open Data Commons Public Domain Dedication and Licence (PDDL)",
    "value": "https://opendatacommons.org/licenses/pddl/1.0/"
  },
  {
    "label": "Open Data Commons Attribution License (ODC-By)",
    "value": "https://opendatacommons.org/licenses/by/1.0/"
  },
  {
    "label": "Open Data Commons Open Database License (ODbL)",
    "value": "https://opendatacommons.org/licenses/odbl/1.0/"
  },
  {
    "label": "Creative Commons Attribution (CC BY)",
    "value": "https://creativecommons.org/licenses/by/4.0/"
  },
  {
    "label": "Creative Commons Attribution-ShareAlike (CC BY-SA)",
    "value": "https://creativecommons.org/licenses/by-sa/4.0/"
  },
  {
    "label": "GNU Free Documentation License",
    "value": "https://www.gnu.org/licenses/fdl-1.3.en.html"
  },

  {
    "label": "Other",
    "value": "other"
  }
]
