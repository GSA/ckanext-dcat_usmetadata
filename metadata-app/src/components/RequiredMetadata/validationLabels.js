export default {
  title: 'Title',
  contactPoint: 'Contact Name',
  identifier: 'Unique ID',
  contactEmail: 'Contact Email',
  description: 'Description',
  publisher: 'Publisher',
  accessLevel: 'Public Access Level',
  dataQuality: 'Access Level',
  rights: 'Rights',
  spatial: 'Spatial',
  license: 'License',
  temporal: 'Temporal',
};
