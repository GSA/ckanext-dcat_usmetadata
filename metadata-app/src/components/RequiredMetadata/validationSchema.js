import * as yup from 'yup';

export default yup.object().shape({
  saveDraft: yup.boolean(),
  title: yup.string().required('Title is required'),
  owner_org: yup.string().required('Organization is required'),
  contact_name: yup.string().when('saveDraft', (saveDraft, schema) => {
    return saveDraft ? schema : schema.required('Contact is required');
  }),
  unique_id: yup.string().when('saveDraft', (saveDraft, schema) => {
    return saveDraft ? schema : schema.required('Unique ID is required');
  }),
  contact_email: yup.string().when('saveDraft', (saveDraft, schema) => {
    return saveDraft
      ? schema
      : schema.email('Must be valid email').required('Contact email is required');
  }),
  description: yup.string().when('saveDraft', (saveDraft, schema) => {
    return saveDraft ? schema : schema.required('Description is required');
  }),
  tags: yup.array().when('saveDraft', (saveDraft, schema) => {
    return saveDraft
      ? schema
      : schema
          .of(
            yup.object().shape({
              name: yup
                .string()
                .min(2)
                .max(100)
                .matches(/[a-zA-Z0-9-_.]+/),
              vocabulary_id: yup.string(),
            })
          )
          .required(
            'Tags are required. Please, provide at least one tag. Note that a tag name must be a string between 2 and 100 characters long containing only alphanumeric characters and -, _ and .'
          );
  }),
  publisher: yup.string().when('saveDraft', (saveDraft, schema) => {
    return saveDraft ? schema : schema.required('Publisher is required');
  }),
  public_access_level: yup.string().when('saveDraft', (saveDraft, schema) => {
    return saveDraft ? schema : schema.required('Access level is required');
  }),
  rights_desc: yup
    .string()
    .required('Please add explanation of rights')
    .when('access_level_comment', (accessLevelComment, schema) => {
      return accessLevelComment === 'false' ? schema : yup.string();
    }),
  spatial: yup
    .string()
    .test('spatial-location-extra', 'Please provide location description.', function validate(
      value
    ) {
      const formVals = this.from[0].value;
      if (value === 'true') {
        if (!formVals.spatial_location_desc) return false;
      }
      return true;
    }),
  spatial_location_desc: yup.string(),
  licenseOther: yup
    .string()
    .required('License is required')
    .url('License should be a valid url')
    .when('license', (license, schema) => {
      return license === 'other' ? schema : yup.string();
    }),
  license: yup.string().when('saveDraft', (saveDraft, schema) => {
    return saveDraft ? schema : schema.required('License is required');
  }),
  temporal: yup
    .string()
    .test('temporal-start-end', 'Please specify start and end date', function validate(value) {
      const formVals = this.from[0].value;
      if (value === 'true') {
        if (!formVals.temporal_start_date || !formVals.temporal_end_date) return false;
      }
      return true;
    }),
  temporal_start_date: yup.date().typeError('Please specify a valid start date'),
  temporal_end_date: yup.date().typeError('Please specify a valid end date'),
});
