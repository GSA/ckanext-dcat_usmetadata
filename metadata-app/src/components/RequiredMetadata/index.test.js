import React from 'react';
import { Formik } from 'formik';
import { render } from '@testing-library/react';
import RequiredMetadata from '.';
import '../../css/custom.css';

const Form = () => (
  <Formik initialValues={{}}>
    {(values) => (
      <form>
        <RequiredMetadata
          apiUrl="http://localhost:5000/api/3/action"
          apiKey="123"
          currentStep={1}
          fetchDatasetsOpts="false"
          values={values}
        />
      </form>
    )}
  </Formik>
);

test('Renders RequiredMetadata component', () => {
  const { container } = render(<Form />);
  expect(container).toMatchSnapshot();
});
