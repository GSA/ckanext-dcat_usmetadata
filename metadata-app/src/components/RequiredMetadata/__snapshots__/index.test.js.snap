// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders RequiredMetadata component 1`] = `
<div>
  <form>
    <div
      class="usa-form-custom"
    >
      <section
        class="site-component-section"
        id="section-basic-mega-menu"
      >
        <p
          class="site-text-intro"
        >
          The following fields are required metadata for each dataset in an agency’s inventory (
          <a
            class="usa-link usa-link--external"
            href="https://www.congress.gov/115/plaws/publ435/PLAW-115publ435.pdf"
            rel="noopener noreferrer"
            target="_blank"
          >
            per Section 202 of the OPEN Government Data Act
          </a>
          ). For more information about the form fields, consult the
           
          <a
            class="usa-link usa-link--external"
            href="https://resources.data.gov/resources/dcat-us/"
            rel="noopener noreferrer"
            target="_blank"
          >
            DCAT-US Schema
          </a>
          .
        </p>
      </section>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
            >
              Title
            </label>
            <div
              class="usa-helptext "
            >
              <span
                class="usa-helptext"
              >
                Use
                 
                <a
                  class="usa-link usa-link--external"
                  href="https://plainlanguage.gov/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  everyday language
                </a>
                 
                to make the dataset easy to find and understand.
              </span>
            </div>
            
            <input
              class="usa-input  undefined undefined"
              name="title"
              placeholder=""
              type="string"
              value=""
            />
          </div>
          <div
            class=""
          >
            <div
              class="wrapper"
            >
              <label
                class="usa-label"
              />
              <div
                class="usa-helptext "
              />
              
              <input
                class="usa-input  undefined undefined"
                name="url"
                style="display: none;"
                type="url"
                value=""
              />
            </div>
            <span
              class="dataset_url"
              style="display: inline;"
            >
              http://localhost/dataset/
            </span>
            <button
              class="usa-button dataset_url_edit"
              type="button"
            >
              Edit
            </button>
          </div>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="wrapper"
        >
          <label
            class="usa-label"
          >
            Description
          </label>
          <div
            class="usa-helptext "
          >
            <span
              class="usa-helptext"
            >
              Write a description (like an abstract) with enough detail to help a user quickly decide if the asset is of interest. You can use
               
              <a
                class="usa-link usa-link--external"
                href="https://www.markdownguide.org/basic-syntax/"
                rel="noopener noreferrer"
                target="_blank"
              >
                Markdown Formatting
              </a>
               
              here.
            </span>
          </div>
          
          <textarea
            class="usa-input  undefined undefined"
            name="description"
            rows="6"
            type="string"
          />
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <span
          class="usa-label"
        >
          Tags
        </span>
        <div
          class="react-tags-input grid-col-12 undefined"
        >
          <span
            class="usa-helptext"
          >
            Use both technical and non-technical terms to help users find your dataset. Press tab or enter to add each new tag.
          </span>
          <div
            class="react-tags"
          >
            <div
              aria-live="polite"
              aria-relevant="additions removals"
              class="react-tags__selected"
            />
            <div
              class="react-tags__search"
            >
              <div
                class="react-tags__search-wrapper"
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded="false"
                  aria-label="Start typing to search"
                  aria-owns="ReactTags"
                  class="react-tags__search-input"
                  placeholder="Start typing to search"
                  role="combobox"
                  style="width: 2px;"
                  value=""
                />
                <div
                  style="position: absolute; width: 0px; height: 0px; visibility: hidden; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                >
                  Start typing to search
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="wrapper"
        >
          <label
            class="usa-label"
          >
            Organization
          </label>
          <div
            class="usa-helptext "
          />
          
          <select
            choices=""
            class="usa-select  "
            name="owner_org"
            type="select"
          >
            <option
              value=""
            >
              -Select-
            </option>
          </select>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <span
            class="usa-label"
          >
            Select Publisher
            <div
              class="tooltip"
            >
              <svg
                height="20px"
                style="margin-left: .5em;"
                tabindex="0"
                width="20px"
              >
                info.svg
              </svg>
            </div>
          </span>
          <div
            class="react-tags-input grid-col-12 undefined"
          >
            <div
              class="usa-helptext"
            >
              <span
                class="usa-helptext"
              >
                Start typing to see list of publishers. If you do not see the Publisher for your dataset listed, please contact
                 
                <a
                  class="usa-link usa-link--external"
                  href="mailto:inventory-help@gsa.gov"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  inventory-help@gsa.gov
                </a>
                 
                for further assistance.
              </span>
            </div>
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-owns="react-autowhatever-1"
              class="react-autosuggest__container"
              role="combobox"
            >
              <input
                aria-autocomplete="list"
                aria-controls="react-autowhatever-1"
                autocomplete="off"
                class="usa-input"
                placeholder="Select publisher"
                type="text"
                value=""
              />
              <div
                class="react-autosuggest__suggestions-container"
                id="react-autowhatever-1"
                role="listbox"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="wrapper"
        >
          <label
            class="usa-label"
          >
            Contact Name
            <div
              class="tooltip"
            >
              <svg
                height="20px"
                style="margin-left: .5em;"
                tabindex="0"
                width="20px"
              >
                info.svg
              </svg>
            </div>
          </label>
          <div
            class="usa-helptext "
          />
          
          <input
            class="usa-input  undefined undefined"
            name="contact_name"
            type="string"
            value=""
          />
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="wrapper"
        >
          <label
            class="usa-label"
          >
            Contact Email
          </label>
          <div
            class="usa-helptext "
          />
          
          <input
            class="usa-input  undefined undefined"
            name="contact_email"
            type="string"
            value=""
          />
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="wrapper"
        >
          <label
            class="usa-label"
          >
            Unique ID
            <div
              class="tooltip"
            >
              <svg
                height="20px"
                style="margin-left: .5em;"
                tabindex="0"
                width="20px"
              >
                info.svg
              </svg>
            </div>
          </label>
          <div
            class="usa-helptext "
          >
            <span
              class="usa-helptext"
            >
              Every dataset must have a ID number that is unique within the agency.
            </span>
          </div>
          
          <input
            class="usa-input  undefined undefined"
            name="unique_id"
            type="string"
            value=""
          />
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="wrapper"
        >
          <label
            class="usa-label"
          >
            Public Access level
          </label>
          <div
            class="usa-helptext "
          />
          
          <select
            choices="[object Object],[object Object],[object Object]"
            class="usa-select  "
            name="public_access_level"
            type="select"
          >
            <option
              value=""
            >
              -Select-
            </option>
            <option
              value="public"
            >
              Public - this dataset can be made publicly available to all without restrictions
            </option>
            <option
              value="restricted public"
            >
              Restricted Public - this dataset is available under certain use restrictions
            </option>
            <option
              value="non-public"
            >
              Non-Public - this dataset is not available to members of the public
            </option>
          </select>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="wrapper"
        >
          <label
            class="usa-label"
          >
            License
          </label>
          <div
            class="usa-helptext "
          />
          
          <select
            choices="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
            class="usa-select  "
            name="license"
            type="select"
          >
            <option
              value=""
            >
              -Select-
            </option>
            <option
              value="n/a"
            >
              N/A - My dataset doesn’t have a license
            </option>
            <option
              value="https://opensource.org/licenses/MIT-0"
            >
              MIT
            </option>
            <option
              value="https://www.usa.gov/publicdomain/label/1.0/"
            >
              U.S. Government Works
            </option>
            <option
              value="https://creativecommons.org/publicdomain/zero/1.0/"
            >
              Creative Commons Zero Public Domain Dedication (CC0)
            </option>
            <option
              value="https://opendatacommons.org/licenses/pddl/1.0/"
            >
              Open Data Commons Public Domain Dedication and Licence (PDDL)
            </option>
            <option
              value="https://opendatacommons.org/licenses/by/1.0/"
            >
              Open Data Commons Attribution License (ODC-By)
            </option>
            <option
              value="https://opendatacommons.org/licenses/odbl/1.0/"
            >
              Open Data Commons Open Database License (ODbL)
            </option>
            <option
              value="https://creativecommons.org/licenses/by/4.0/"
            >
              Creative Commons Attribution (CC BY)
            </option>
            <option
              value="https://creativecommons.org/licenses/by-sa/4.0/"
            >
              Creative Commons Attribution-ShareAlike (CC BY-SA)
            </option>
            <option
              value="https://www.gnu.org/licenses/fdl-1.3.en.html"
            >
              GNU Free Documentation License
            </option>
            <option
              value="other"
            >
              Other
            </option>
          </select>
        </div>
        <div
          class="wrapper"
        >
          <label
            class="usa-label"
          />
          <div
            class="usa-helptext disabled"
          >
            <span
              class="usa-helptext"
            >
              If you selected “Other”, please specify the name of your License in URL format. Please include "https://" at the beginning of your URL.*
            </span>
          </div>
          
          <input
            class="usa-input disabled undefined undefined"
            disabled=""
            name="licenseOther"
            type="string"
            value=""
          />
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <span
            class="usa-label"
          >
            Rights
          </span>
           
          <br />
          <div
            class="form-group "
          >
            <input
              class="usa-radio__input"
              id="rights_option_1"
              name="access_level_comment"
              type="radio"
              value="true"
            />
            <label
              class="usa-radio__label"
              for="rights_option_1"
            >
              My dataset is public
            </label>
          </div>
          <div
            class="form-group "
          >
            <input
              class="usa-radio__input"
              id="rights_option_2"
              name="access_level_comment"
              type="radio"
              value="false"
            />
            <label
              class="usa-radio__label"
              for="rights_option_2"
            >
              My dataset is not public
            </label>
          </div>
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
            />
            <div
              class="usa-helptext "
            >
              <span
                class="usa-helptext"
              >
                If your dataset is not public, please add an explanation of rights and feel free to include any instructions on restrictions, or how to access a restricted file (max 255 characters)*
              </span>
            </div>
            
            <input
              class="usa-input  undefined undefined"
              name="rights_desc"
              type="string"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <span
            class="usa-label"
          >
            Relevant Location
          </span>
           
          <br />
          <div
            class="form-group "
          >
            <input
              class="usa-radio__input"
              id="spatial_option_1"
              name="spatial"
              type="radio"
              value="false"
            />
            <label
              class="usa-radio__label"
              for="spatial_option_1"
            >
              My dataset does not have a spatial component
            </label>
          </div>
          <div
            class="form-group "
          >
            <input
              class="usa-radio__input"
              id="spatial_option_2"
              name="spatial"
              type="radio"
              value="true"
            />
            <label
              class="usa-radio__label"
              for="spatial_option_2"
            >
              My dataset does have a spatial component
            </label>
          </div>
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
            />
            <div
              class="usa-helptext "
            >
              <span
                class="usa-helptext"
              >
                If your dataset has a spatial component, please provide location such as place name or latitude/longitude pairs above*
              </span>
            </div>
            
            <input
              class="usa-input  undefined undefined"
              name="spatial_location_desc"
              type="string"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <span
            class="usa-label"
          >
            Temporal
            <div
              class="tooltip"
            >
              <svg
                height="20px"
                style="margin-left: .5em;"
                tabindex="0"
                width="20px"
              >
                info.svg
              </svg>
            </div>
          </span>
           
          <br />
          <div
            class="form-group "
          >
            <input
              class="usa-radio__input"
              id="temporal_option_1"
              name="temporal"
              type="radio"
              value="false"
            />
            <label
              class="usa-radio__label"
              for="temporal_option_1"
            >
              My dataset does not have a start and end date for the applicability of data
            </label>
          </div>
          <div
            class="form-group "
          >
            <input
              class="usa-radio__input"
              id="temporal_option_2"
              name="temporal"
              type="radio"
              value="true"
            />
            <label
              class="usa-radio__label"
              for="temporal_option_2"
            >
              My dataset has a start and end date for the applicability of data
            </label>
          </div>
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
              for="temporal_start_date"
            />
            <div
              class="usa-helptext "
            >
              <span
                class="usa-helptext"
              >
                If your dataset has a temporal component, please provide start date for applicability of data above (MM/DD/YYYY)*
              </span>
            </div>
            
            <div>
              <input
                class="usa-input "
                id="temporal_start_date"
                name="temporal_start_date"
                type="date"
                value=""
              />
            </div>
          </div>
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
              for="temporal_end_date"
            />
            <div
              class="usa-helptext "
            >
              <span
                class="usa-helptext"
              >
                If your dataset has a temporal component, please provide end date for applicability of data above (MM/DD/YYYY)*
              </span>
            </div>
            
            <div>
              <input
                class="usa-input "
                id="temporal_end_date"
                name="temporal_end_date"
                type="date"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="margin-top-6 clearfix"
      >
        <button
          class="usa-button usa-button--outline"
          type="button"
        >
          Save draft
        </button>
        <button
          class="usa-button float-right margin-right-0"
          type="button"
        >
          Save and Continue
        </button>
      </div>
    </div>
  </form>
</div>
`;
